Trappist/src/Promact.Trappist.Test/Category/CategoryRepositoryTest.cs

```diff
@@ -111,5 +133,75 @@ private DomainModel.Models.Category.Category CreateCategory()
             };
             return category;
         }
+
+        /// <summary>
+        /// Creating single answer Question
+        /// </summary>
+        /// <returns>Object of single answer Question</returns>
+        private async Task<QuestionAC> CreateSingleAnswerQuestionAsync(int categoryId, string userId)
+        {
+            var singleAnswerQuestion = new QuestionAC
+            {
+                Question = new QuestionDetailAC()
+                {
+                    QuestionDetail = "what is git?",
+                    CategoryID = categoryId,
+                    DifficultyLevel = DifficultyLevel.Hard,
+                    QuestionType = QuestionType.Single
+                },
+                SingleMultipleAnswerQuestion = new SingleMultipleAnswerQuestionAC()
+                {
+                    SingleMultipleAnswerQuestion = new SingleMultipleAnswerQuestion(),
+                    SingleMultipleAnswerQuestionOption = new List<SingleMultipleAnswerQuestionOption>()
+                    {
+                        new SingleMultipleAnswerQuestionOption()
+                        {
+                            IsAnswer = true,
+                            Option = "distributed version control system",
+                        },
+                        new SingleMultipleAnswerQuestionOption()
+                        {
+                            IsAnswer = false,
+                            Option = "continuous integration service",
+                        },
+                        new SingleMultipleAnswerQuestionOption()
+                        {
+                            IsAnswer = false,
+                            Option = "Collaborations system",
+                        }
+                    }
+                }
+            };
+            await _questionRepository.AddSingleMultipleAnswerQuestionAsync(singleAnswerQuestion, userId);
+            return singleAnswerQuestion;
+        }
+
+        /// <summary>
+        /// Method to create new user
+        /// </summary>
+        /// <param name="userName">User name</param>
+        /// <returns>User object</returns>
+        public async Task<ApplicationUser> CreateUserAsync(string userName)
+        {
+            ApplicationUser user = new ApplicationUser() { Email = userName, UserName = userName };
+            await _userManager.CreateAsync(user);
+            return await _userManager.FindByEmailAsync(user.Email);
+        }
+
+        /// <summary>
+        /// Method to create new Category
+        /// </summary>
+        /// <param name="categoryName">Category name</param>
+        /// <returns>Category object</returns>
+        public async Task<DomainModel.Models.Category.Category> CreateCategoryAsync(string categoryName)
+        {
+            var category = new DomainModel.Models.Category.Category
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename 'category' which hides the field with the same name. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1117)